src/core/Label/Label.tsx

```diff
@@ -0,0 +1,73 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import React from 'react';
+import cx from 'classnames';
+import { useTheme } from '../utils';
+import '@itwin/itwinui-css/css/inputs.css';
+
+type LabelOwnProps<T extends React.ElementType = 'label'> = {
+  /**
+   * What element should the label be rendered as?
+   * @default 'label'
+   */
+  as?: T;
+  /**
+   * Set the display style of the label.
+   *   - 'block' - default, no extra spacing (label and input will be on separate lines)
+   *   - 'inline' - label gets a right margin so it can be placed inline
+   * @default 'block'
+   */
+  displayStyle?: 'block' | 'inline';
```

Comment:
I am not sure about this prop name. Maybe it should just be called `inline` or something. Suggestions welcome

Reply:
In CSS it's just display, maybe that's enough?

Reply:
Well, the problem is, it doesn't really change the actual `display` value (it stays `inline-block` for both cases). It just changes the margin

I only called it `displayStyle` because that's the name we use in our inputs

Reply:
I vote for consistency. If we already use it in inputs, lets use same name.

Reply:
Shouldn't it be `display: block` to take all width when it is `block` type?

Reply:
span in pink
![image](https://user-images.githubusercontent.com/9084735/141806014-0ddac842-603b-41b5-b93c-c32e69b24db4.png)

Reply:
Fixed CSS and bumped version